.github/workflows/ci.yml

```
@@ -23,14 +25,12 @@ jobs:
           - os: macos-latest
             arch: armv7
           - os: macos-latest
-            arch: x86
-            
+            arch: x86        
     steps:
       - uses: actions/checkout@v2
-      - name: Build and run tests
-        run: make && make test
-
-  build-msys2:
+      - run: make
+      - run: make test
+  windows:
     runs-on: windows-latest
```

If you want you can add
```suggestion
    runs-on: windows-latest
    defaults:
      run:
        shell: msys2 {0}
```
And avoid duplicating it between the two steps below

---

I thought about that. But for the first step, which sets up the MSYS2 environment, I figured we actually don't want to run that step inside the `msys2` shell. So I figured we didn't want it as a job-wide default.

---

We do that over at msys2, and it works fine.  That default shell is only used by `run` steps, not actions (which are actually run in node,js)

---

Oh interesting, I had always assumed that the default shell would also be used by actions.

I wonder what happens in composite actions (i.e. non-Javascript actions).

---

Let's give it a try - https://github.com/JuliaMath/openlibm/pull/247

---

I don't know.  Composite actions are newer than the msys2 CIs I've dealt with

---

This is getting pretty far afield, but https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsshell says shell is *required* for run steps in a composite action, so I guess the default shell doesn't apply.